homeassistant/components/ping/entity.py

```diff
@@ -0,0 +1,26 @@
+"""Base entity for Ping integration."""
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import PingUpdateCoordinator
+
+
+class BasePingEntity(CoordinatorEntity[PingUpdateCoordinator]):
+    """Representation of a Ping base entity."""
+
+    def __init__(
+        self, config_entry: ConfigEntry, coordinator: PingUpdateCoordinator
+    ) -> None:
+        """Initialize the Ping Binary sensor."""
+        super().__init__(coordinator)
+
+        self._attr_name = config_entry.title
```

Comment:
`_attr_has_entity_name = True` and `_attr_name=None`? Maybe need to check how this works with the device tracker entity

Reply:
But this will not rename the entities when renaming the config entry. It was suggested in https://github.com/home-assistant/core/pull/103743#discussion_r1391335908 not to use a "name" option in the config flow, instead use the config entry title.

Reply:
What @joostlek is proposing is to use the device name instead of just the entity name.
I don't know either if this works well on this case but we should strive for implementing `_attr_has_entity_name` on as many as possible to go for devices and its name in case of config flows.

Reply:
I fully agree and I'm fine with this change, it was just a note that the user then has to change the title of the config entry and the device name, but that is something the user already knows.